src/main/java/com/github/manolo8/darkbot/backpage/GalaxyManager.java

```diff
@@ -0,0 +1,107 @@
+package com.github.manolo8.darkbot.backpage;
+
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.backpage.entities.galaxy.*;
+import org.dom4j.Document;
+import org.dom4j.Element;
+import org.dom4j.io.SAXReader;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GalaxyManager {
+
+    public BackpageManager backpageManager;
+    public Jumpgate jumpgate;
+    private Main main;
+
+    public GalaxyManager(Main main, BackpageManager backpageManager) {
+        this.main = main;
+        this.backpageManager = backpageManager;
+        this.jumpgate = new Jumpgate();
+    }
+
+    private static Gate analyzeGate(Element e) {
+        Gate gate = new Gate();
+
+        gate.setCurrent(Integer.parseInt(e.attributeValue("current")));
+        gate.setCurrentWave(Integer.parseInt(e.attributeValue("currentWave")));
+        gate.setId(Integer.parseInt(e.attributeValue("id")));
+        gate.setLifePrice(Integer.parseInt(e.attributeValue("lifePrice")));
+        gate.setLivesLeft(Integer.parseInt(e.attributeValue("livesLeft")));
+        gate.setPrepared(Integer.parseInt(e.attributeValue("prepared")));
+        gate.setState(e.attributeValue("state"));
+        gate.setTotal(Integer.parseInt(e.attributeValue("total")));
+        gate.setTotalWave(Integer.parseInt(e.attributeValue("totalWave")));
+        return gate;
+    }
+
+    private static EnergyCost analyzeEnergyCost(Element e) {
+        EnergyCost energyCost = new EnergyCost();
+
+        energyCost.setMode(e.attributeValue("mode"));
+        energyCost.setValue(Integer.parseInt(e.getText()));
+        return energyCost;
+    }
+
+
+    private static Multiplier analyzeMultipler(Element e) {
+        Multiplier multiplier = new Multiplier();
+
+        multiplier.setMode(e.attributeValue("mode"));
+        multiplier.setState(Integer.parseInt(e.attributeValue("state")));
+        multiplier.setValue(Integer.parseInt(e.attributeValue("value")));
+        return multiplier;
+    }
+
+    public void updateJumpgateInfo() {
+        String params = "flashinput/galaxyGates.php?userID=" + main.hero.id + "&action=init&sid=" + main.statsManager.sid;
+        updateJumpgateInfo(params, 100);
+    }
+
+    public void updateJumpgateInfo(String params, int minWait) {
```

Comment:
`"flashinput/galaxyGates.php?userID=" + main.hero.id + "&action=init&sid=" + main.statsManager.sid` shouldn't need to be a parameter. If it's always part of the request, you shouldn't need to send it. I'm guessing you intend this parameter to allow for spinning, but then you should just make 2 different (public) calls that end here, and make this method private